### DISCUSSION TITLE
arm

### DISCUSSION THREAD

**Discussion by wdthompson** [CatalystDevOrg/Catalyst#445] [General]
It is REALLY hard to find an arm64 deb, .......and
it is only 3.4
dmg does not help if you don't have a mac
Please change name .. arm64 ... to aarch64
Many times I have downloaded amd64 by mistake, yes I should pay more attention
error msg .. repeated several times
[15791:1001/133425.839433:ERROR:gbm_wrapper.cc(255)] Failed to export buffer to dma_buf: No such file or directory (2)
but since this is some versions back, not of much value

**Comment by jdev082** [comment]
I do not have control over what the build system names the files. Can I have more details on the error?

**Comment by wdthompson** [comment]
No other errors, repeated 16 times, only the timestamps varies, running on most recent devuan
uname -a
Linux rockpro64 6.1.34 #1 SMP Sat Jun 17 20:38:13 EDT 2023 aarch64 GNU/Linux
This on a rockpro, rk3399

**Reply by jdev082** [reply]
So does the latest build of Catalyst open and run properly on your device?

**Comment by wdthompson** [comment]
Under 3.5.3 there is no arm64.deb only arm64.dmg
BTW, it does run, maybe I was unclear (3.4)

**Reply by jdev082** [reply]
Adding arm64 deb support to the build system will take some time. For the mean time, you can manually build an arm64 deb of Catalyst on the device itself. You'll need a copy of nodejs installed. Here's the build instructions:

```
git clone -b v3.5.3 https://github.com/CatalystDevOrg/Catalyst
cd Catalyst
npm install
npm run build
npm run make
```
then in out/make there should be the correct package in one of the folders

**Comment by wdthompson** [comment]
OK, all went well until the last step
-----
An unhandled rejection has occurred inside Forge:
Error: Cannot make for rpm, the following external binaries need to be installed: rpmbuild
at MakerRpm.ensureExternalBinariesExist (/home/d/data/Catalyst/node_modules/@electron-forge/maker-base/dist/Maker.js:103:19)
    at Task.task (/home/d/data/Catalyst/node_modules/@electron-forge/core/dist/api/make.js:118:27)
    at Task.run (/home/d/data/Catalyst/node_modules/listr2/dist/index.cjs:978:35)
----
apt install rpmbuild didn't do anything, I have no idea what it would be called
or if it even exists in a debian system

**Reply by jdev082** [reply]
All it does is check for the binary, and isn't essential to build a deb. You can just run this:

```
sudo touch /usr/bin/rpmbuild
```
that creates an empty file named rpmbuild at the path it checks. do that and it will build just fine.

**Reply by wdthompson** [reply]
and I get same error message, identical, maybe that path is not right?
...tried /usr/sbin,,, /bin,,, /sbin and all same error msg
could it be that npm run build has to be redone?
And it has to be executable,, 755

**Reply by jdev082** [reply]
is it still asking for rpmbuild or a different binary?

**Reply by wdthompson** [reply]
and it works,, seems no easy way to increase font size ??
errors, same as before, plus...
[22337:1001/150738.034329:ERROR:gbm_wrapper.cc(253)] Failed to export buffer to dma_buf: No such file or directory (2)
[22337:1001/150804.223264:ERROR:shared_image_manager.cc(217)] SharedImageManager::ProduceSkia: Trying to Produce a Skia representation from a non-existent mailbox.
[22337:1001/151126.422608:ERROR:gl_surface_presentation_helper.cc(260)] GetVSyncParametersIfAvailable() failed for 1 times!
[22307:1001/151148.038980:ERROR:CONSOLE(1)] "Uncaught (in promise) TypeError: Failed to fetch", source: devtools://devtools/bundled/panels/elements/elements.js (1)
........
until chmod 755 /usr/bin/rpmbuild,, identical error msgs

**Reply by jdev082** [reply]
The errors in terminal are normal. Font scaling hasn't been implemented yet.

**Reply by wdthompson** [reply]
Thanks for your help, for us older guys tiny fonts are a bit painful,,
I have reading glasses for micro soldering, don't like wearing for computer.
Don't get me wrong, it is just marginally too small, not ridiculous tiny
..... btw, if you don't have an easy way to make a deb would you like me to send it somewhere?

**Reply by jdev082** [reply]
I am working on an initial implementation of font scaling and might be able to ship functional font scaling in v3.6.0
